src/Rings/FreeAssociativeAlgebraIdeal.jl

```diff
@@ -184,37 +193,65 @@ Ideal generating system with elements
 function groebner_basis(I::FreeAssociativeAlgebraIdeal,
   deg_bound::Int=-1;
   protocol::Bool=false,
-  interreduce::Bool=false)
+  interreduce::Bool=false,
+  algorithm::Symbol=:f4
+  )
   isdefined(I, :gb) && (I.deg_bound == -1 || I.deg_bound >= deg_bound) && return I.gb
-  I.gb = groebner_basis(IdealGens(gens(I)), deg_bound; ordering=:deglex, protocol=protocol, interreduce=interreduce)
+  I.gb = groebner_basis(IdealGens(gens(I)), deg_bound; ordering=:deglex, protocol=protocol, interreduce=interreduce, algorithm=algorithm)
   I.deg_bound = deg_bound
   return I.gb
 end
 function groebner_basis(g::IdealGens{<:FreeAssociativeAlgebraElem},
   deg_bound::Int=-1;
   ordering::Symbol=:deglex,
   protocol::Bool=false,
-  interreduce::Bool=false)
-  gb = groebner_basis(collect(g), deg_bound; ordering=ordering, protocol=protocol, interreduce=interreduce)
+  interreduce::Bool=false,
+  algorithm::Symbol=:f4
+  )
+  gb = groebner_basis(collect(g), deg_bound; ordering=ordering, protocol=protocol, interreduce=interreduce,algorithm=algorithm)
+  gb = Generic.FreeAssociativeAlgebraElem{QQFieldElem}[x for x in gb]
   return IdealGens(gb)
 end
 function groebner_basis(g::Vector{<:FreeAssociativeAlgebraElem},
   deg_bound::Int=-1;
   ordering::Symbol=:deglex,
   protocol::Bool=false,
-  interreduce::Bool=false)
+  interreduce::Bool=false,
+  algorithm::Symbol=:f4)
 
+  @req length(g) > 0 "Vector must not be empty"
   R = parent(g[1])
   @req all(x -> parent(x) == R, g) "parent mismatch"
-  @req deg_bound >= 0 || !protocol "computing with a protocol requires a degree bound"
+  #@req deg_bound >= 0 || !protocol "computing with a protocol requires a degree bound"
   @req ordering == :deglex || deg_bound > 0 "only :deglex ordering is supported for no degree bound"
+  @req algorithm in (:f4, :buchberger, :letterplace) "Not a valid algorithm, use :f4ncgb, :buchberger or :letterplace"
+
+  if algorithm == :f4
+    @req ordering == :deglex "f4 only supports :deglex ordering"
+    @req R.base_ring == QQ "only rational coefficients are supported"
```

Comment:
```suggestion
    @req base_ring(R) == QQ "only rational coefficients are supported"
```
Then perhaps it would be better to not have this as default unless we really are over the rationals?